source/plugin/game-profile-manager.rst

```diff
@@ -0,0 +1,138 @@
+
+To retrieve a ``GameProfile`` from a ``CompletableFuture<GameProfile``, you can simply call the ``CompletableFuture#get``
+method.
+
+.. code-block:: java
+
+    GameProfile gameProfile = futureGameProfile.get();
+
+.. warning::
+    If the ``GameProfile`` is not immediately available (such as if the cache is not being used or does not contain the
+    ``GameProfile``), then ``get`` will wait for the future to complete. For that reason, it is not advisable to use
+    this on the main thread as it will halt the server. Alternatively, you can use the
+    ``CompletableFuture#thenAccept(Consumer<? super T>)`` method to specify a ``Consumer`` to be run upon completion.
+
+Creating GameProfiles
+=====================
+
+You can generate a new ``GameProfile`` using ``GameProfileManager#createProfile(UUID, String)``. Note that the username
+does not necessarily need to correspond to the ``UUID`` of that player. Likewise, the ``UUID`` does not need to belong
+to a valid player.
+
+.. code-block:: java
+
+    GameProfile gameProfile =
+        profileManager.createProfile(
+            UUID.fromString("00000000-0000-0000-0000-000000000000"),
+            "Herobrine");
+
+.. note::
+    It is not mandatory to specify the name of the ``GameProfile`` (``null`` is a valid argument).
+
+Filling GameProfiles
+====================
+
+Filling a ``GameProfile`` completes the profile by fetching information like the player's skin from the Mojang API.
+Note that if faked data like username is associated with a certain UUID, it will be replaced by the actual data from
+the Mojang API.
+
+.. code-block:: java
+
+    GameProfile filledProfile = profileManager.fill(gameProfile).get();
+
+Working with ProfileProperties
+==============================
+
+``GameProfile``\ s can be used to store arbitrary data about a player using ``ProfileProperty``\ s. However, this cannot
+not be used as a permanent data store, as the data is not persisted across server restarts. We can retrieve the properties
+of a ``GameProfile`` using the ``GameProfile#getPropertyMap`` method, which returns a ``Multimap``. From there, you can
+retrieve existing or store new ``ProfileProperty``\ s, which are represented as a key value pair. To generate a new
+``ProfileProperty``, simply call the ``GameProfileManager#createProfileProperty(String, String, @Nullable String)``
+method. The third argument (signature) is optional. However, a valid signature from Mojang must be specified for certain
+properties.
+
+.. code-block:: java
+
+    import org.spongepowered.api.profile.property.ProfileProperty;
+
+    import java.util.Collection;
+
+    profile.getPropertyMap().put(
+        "key", profileManager.createProfileProperty("foo", "bar", null));
+    Collection<ProfileProperty> customProperties = profile.getPropertyMap().get("key");
+
+Working with the GameProfileCache
+=================================
+
```

Comment:
It should be documented that `GameProfileCache` can be replaced by plugins, as well.